src/mate_clj/core.clj

```diff
@@ -202,3 +202,11 @@
 (defn dsplit-with
   [pred coll]
   [(dtake-while pred coll) (ddrop-while pred coll)])
+
+(defmacro dwhile
+  [test & body]
+  `(loop []
+     (println ~@test "=>" ~test)
```

Comment:
When I change this line to:
```clojure
(println '~test "=>" ~test)
```
It is working with both examples.
The reason from my understanding is that if you use a macro as `test` you want to read it, but not eval it so the unquote - `~` try to evaluate it (in this case `or`) but the quote macro - `'` will prevent its evaluation so we can print it as is. This happens because the whole `dwhile` function implementation is a macro so we have to play with it a little bit with the unquote and quote of other macros inside.

Reply:
thanks for the quick response and the explanation